### PR TITLE
Resolve nested url can't change

### PR DESCRIPTION
**Pull request by SergeShkurko**
Added default path ("/"). It resolve problem with subroute data change (#224 )

Before "Nested component" inactive, already - active

For example:
url: 'localhost:8080/foo'

***Before:***

```html
First component:
<app-location route="{{route}}"></app-location>
<app-route route="{{route}}" pattern="/:page" tail="{{subroute}}"></app-route>

Nested component:
<app-route route="{{subroute}}" pattern="/:subpage"></app-route>
```

```js
// Route:
{
  prefix: '',
  path: '/foo'
}

// Tail
{
  prefix: '/foo',
  path: '' // default path == '' & subroute not active 😿
}
```

***After***

```html
First component:
<app-location route="{{route}}"></app-location>
<app-route route="{{route}}" pattern="/:page" tail="{{subroute}}"></app-route>

Nested component (Added flag "auto-activate"):
<app-route auto-activate route="{{subroute}}" pattern="/:subpage"></app-route>
```

```js
// Route:
{
  prefix: '',
  path: '/foo'
}

// Tail
{
  prefix: '/foo',
  path: '/' // default path == '/' & subroute active
}
```

### CODE REVIEW

**Comment by SergeShkurko** [comment]
I signed it!

**Comment by e111077** [comment]
This is a simple solution to a common problem, but the issue is that this is a breaking change. would it be possible to put this behind a flag and also write a test for it?

**Comment by SergeShkurko** [comment]
@e111077 Change is not critical. Tests added

File: test/app-example-2.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+<link rel='import' href='../app-route.html'>
+<link rel='import' href='../app-location.html'>
+
+<dom-module id='app-example-2'>
+  <template>
+    <app-location route='{{route}}'>
+    </app-location>
+    <app-route id="page" route='{{route}}' pattern='/:page' data='{{data}}' tail='{{subroute}}'>
+    </app-route>
+    <app-example-2-subpage route='{{subroute}}'></app-example-2-subpage>
```
[the remark this answers is not in the document]
Comment (SergeShkurko) [reply test/app-example-2.html]: For the accuracy of the result in real conditions
The idea of the tests was taken from issue #224

File: test/test-app-example-2.html [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      exampleApp = fixture('ExampleApp2');
+    });
+
+    teardown(function() {
+      window.history.replaceState({}, '', originalLocation);
+    });
+
+    test('runs through basic usage', function() {
+      // Navigate to /subroutepage
+      setLocation('/subroutepage');
```
[the remark this answers is not in the document]
Comment (SergeShkurko) [reply test/test-app-example-2.html]: Fixed

File: app-route.html [hunk 3]
```diff
@@ -247,6 +247,10 @@
         var path = this.route.path;
         var pattern = this.pattern;
 
+        if (path === '') {
```
[the remark this answers is not in the document]
Comment (SergeShkurko) [reply app-route.html]: Flag added

**Comment by SergeShkurko** [comment]
@e111077 edits made

File: app-route.html [hunk 4]
```diff
@@ -151,6 +151,14 @@
           readOnly: true
         },
 
+        /**
+         * Auto activate route if path empty
+         */
+        "auto-activate": {
```
Comment (e111077) [inline app-route.html]: Please change to camelCase.

Polymer should automatically do kebab case to camel case conversion. It should be fine to call this `autoActivate` as Polymer will translate `<element auto-activate></element>` to `element.autoActivate`.

File: app-route.html [hunk 5]
```diff
@@ -247,6 +255,10 @@
         var path = this.route.path;
         var pattern = this.pattern;
 
+        if (this['auto-activate'] && path === '') {
```
Comment (e111077) [inline app-route.html]: please change to camelCase

File: test/app-example-2.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (e111077) [reply test/app-example-2.html]: 👍

File: app-route.html (hunk 4 above)
Comment (SergeShkurko) [reply app-route.html]: Sorry, I did not know about this possibility of polymer

**Comment by SergeShkurko** [comment]
@e111077 Fixed

**Comment by e111077** [comment]
Very well done, thanks for the PR

**Pull request merged**